### PR TITLE
Gnupg2 Homebrew install on macOS provides gpg command not gpg2.

### PR DESCRIPTION
**Pull request by gartung**
Using Homebrew to install gnupg2 on macOS provides symbolic link /usr/local/bin/gpg but not /usr/local/bin/gpg2. On macOS check for gpg in addition to gpg2.

This might break test where keys are created because gnupg has a limit on the length of the path to the keystore. It may require running
```
spack test --basetemp /tmp/spack
```
 on macOS

### CODE REVIEW

**Comment by adamjstewart** [comment]
You could do something like we do for Python where we search for `python3.7`, then `python3`, then `python` if the first two aren't found. We do something similar for our default `EDITOR` search.

File: lib/spack/spack/util/gpg.py [hunk 1]
```diff
@@ -4,6 +4,7 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import os
+import platform
 
 import spack.paths
 from spack.util.executable import Executable
```
Comment (adamjstewart) [inline lib/spack/spack/util/gpg.py]:
```suggestion
from spack.util.executable import which
```

File: lib/spack/spack/util/gpg.py [hunk 2]
```diff
@@ -32,6 +33,13 @@ class Gpg(object):
     def gpg():
         # TODO: Support loading up a GPG environment from a built gpg.
         gpg = Executable('gpg2')
```
Comment (adamjstewart) [inline lib/spack/spack/util/gpg.py]:
```suggestion
        gpg = which('gpg2', 'gpg')
```

File: lib/spack/spack/util/gpg.py [hunk 3]
```diff
@@ -31,7 +32,14 @@ class Gpg(object):
...
         # TODO: Support loading up a GPG environment from a built gpg.
-        gpg = Executable('gpg2')
+        gpg = which('gpg2', 'gpg')
+        # The gnupg2 Homebrew package install /usr/local/bin/gpg
+        # Requires the use of
+        # spack test --basetemp /tmp/spack
+        # on macOS when creating keys
+        # because $TMPDIR path is too long.
+        if platform.system().lower() == 'darwin':
+            gpg = Executable('gpg')
```
Comment (adamjstewart) [inline lib/spack/spack/util/gpg.py]: You can remove all this additional logic now

File: lib/spack/spack/util/gpg.py [hunk 4]
```diff
@@ -4,9 +4,10 @@
 # SPDX-License-Identifier: (Apache-2.0 OR MIT)
 
 import os
+import platform
```
Comment (adamjstewart) [inline lib/spack/spack/util/gpg.py]: I think this import can be removed to

**Comment by gartung** [comment]
After merging with develop there are no changes.

**Comment by gartung** [comment]
Looks like @tgamblin came up with a better implementation anyway in  #14594

**Pull request closed without merging**